semantic_conventions/trace/messaging.yaml

```
@@ -28,46 +51,114 @@ groups:
         brief: 'The kind of message destination'
-      - id: temp_destination
-        type: boolean
-        requirement_level:
-          conditionally_required: If value is `true`. When missing, the value is assumed to be `false`.
-        brief: 'A boolean that is true if the message destination is temporary.'
-      - id: protocol
+      - id: template
         type: string
-        brief: 'The name of the transport protocol.'
-        examples: ['AMQP', 'MQTT']
-      - id: protocol_version
-        type: string
-        brief: 'The version of the transport protocol.'
-        examples: '0.9.1'
-      - id: url
+        brief: Low cardinality field representing messaging destination
+        note: >
+          Destination names could be constructed from templates.
+          An example would be a destination name involving a user name or product id.
+          Although the destination name in this case is of high cardinality,
+          the underlying template is of low cardinality and can be effectively
+          used for grouping and aggregation.
+        examples: ['/customers/{customerId}']
+      - id: temporary
+        type: boolean
+        brief: 'A boolean that is true if the message destination is temporary and might not exist anymore after messages are processed.'
+      - id: anonymous
+        type: boolean
+        brief: 'A boolean that is true if the message destination is anonymous (could be unnamed or have auto-generated name).'
+
+  - id: messaging.source
+    prefix: messaging.source
+    # todo (https://github.com/open-telemetry/build-tools/issues/123)
+    type: span
+    brief: 'Semantic convention for attributes that describe messaging source on broker'
+    attributes:
+      - id: name
         type: string
-        brief: 'Connection string.'
-        examples: ['tibjmsnaming://localhost:7222', 'https://queue.amazonaws.com/80398EXAMPLE/MyQueue']
-      - id: message_id
+        brief: 'The message source name'
+        note: |
+          Source name SHOULD uniquely identify a specific queue, topic, or other entity within the broker. If
+          the broker does not have such notion, the source name SHOULD uniquely identify the broker.
+        examples: ['MyQueue', 'MyTopic']
+      - id: kind
+        type:
+          allow_custom_values: true
+          members:
+            - id: queue
+              value: "queue"
+              brief: "A message received from a queue"
+            - id: topic
+              value: "topic"
+              brief: "A message received from a topic"
+        brief: 'The kind of message source'
+      - id: template
         type: string
-        brief: 'A value used by the messaging system as an identifier for the message, represented as a string.'
-        examples: '452a7c7c7c7048c2f887f61572b18fc2'
-      - id: conversation_id
+        brief: 'Low cardinality field representing messaging source'
```

```suggestion
        brief: 'Low cardinality representation of the messaging source name'
```
As this only refers to the name, and not to other attributes of the source.